d4s/src/main/scala/d4s/util/OffsetLimit.scala

```diff
@@ -1,10 +1,10 @@
 package d4s.util
 
-import d4s.codecs.{D4SDecoder, D4SEncoder}
+import d4s.codecs.{D4SAttributeEncoder, D4SDecoder, D4SEncoder}
 
 final case class OffsetLimit(offset: Int, limit: Short)
 
 object OffsetLimit {
-  implicit val enc: D4SEncoder[OffsetLimit] = D4SEncoder.derived[OffsetLimit]
+  implicit val enc: D4SAttributeEncoder[OffsetLimit] = D4SAttributeEncoder.derived[OffsetLimit]
   implicit val dec: D4SDecoder[OffsetLimit] = D4SDecoder.derived[OffsetLimit]
 }
```

Comment:
This was made because we now using CastedMagnolia again

Reply:
@neko-kai, then I should revert `CastedMagnolia` also.

Reply:
@vpodlnk Why? Without CastedMagnolia `D4SEncoder.derived[Either[String, Int]]` does not compile.

Reply:
Yup, `CastedMagnolia`  is macros, due to the `OffsetLimit` definition located at the same compilation unit as the `CastedMagnoli` def.

Reply:
I guess the only way is to make utils as a separate module or use `D4SAttributeEncoder`

Reply:
@vpodlnk 
Aaah. got it
You could add
```scala
object D4SEncoder {
  private[d4s] noncastedGen[T]: D4SAttributeEncoder[T] = macro Magnolia.gen[T]
}
```

and then just cast it manually in utils:

```scala
object OffsetLimit {
  implicit val enc: D4SEncoder[OffsetLimit] = D4SEncoder.noncastedGen[OffsetLimit].asInstanceOf[D4SEncoder[OffsetLimit]]
}
```

Reply:
Ah, yup, cool trick.
I'll add.